server/game/cards/challengeevent.js

```diff
@@ -0,0 +1,46 @@
+const DrawCard = require('../drawcard.js');
+
+class ChallengeEvent extends DrawCard {
+    // event card that can only be played during a challenge
+
+    constructor(owner, cardData, challengeType, participatingAs) {
+        // - challengeType (optional): make the event playable only on a
+        //   challenge of the given type
+        //
+        // - participatingAs (optional): make the event playable only when the
+        //   event controller is participating as either 'attacker' or
+        //   'defender'
+        //
+        super(owner, cardData);
+
+        this.challengeType = challengeType;
+        this.participatingAs = participatingAs;
+    }
+
+    canPlay(player, card) {
+        if(player !== this.controller || this !== card) {
+            return false;
+        }
```

Comment:
this needs to go on the base card - that's more a note for myself than to you, though :)